### ISSUE TITLE
Second transaction does not appear after approve? 

### ISSUE THREAD

**Issue by skimaharvey** [PatrickAlphaC/defi-stake-yield-brownie-freecode#7]
Hello,

For some reason the 2nd metasmask transaction does not appear after being approved? 
Any idea where this could be coming from? 
I do have the notification coming "ERC-20 token transfer approved! Now approve the 2nd transaction."
Also I am having the already reported issues but i dont think that is related : 
- wallet does not disconnect 
- "supportedChain is deprecated, please pass networks instead " however it is just a warning not an error (I am able to get the correct chainId

Thank you for your help.

**Comment by PatrickAlphaC** [comment]
I think you're right... I think this repo needs a UI update

**Comment by amanmehra262001** [comment]
Yes its same with me have got any solutions for the same @skimaharvey @PatrickAlphaC 
Thanks in advance

**Comment by amanmehra262001** [comment]
BTW thnq @PatrickAlphaC for amazing free course
also where can i find your more courses

**Comment by PatrickAlphaC** [comment]
https://www.youtube.com/c/PatrickCollins/about

I also have medium articles, chainlink blog articles, twitter, I'm all over the place.

Glad you liked them!

**Comment by amanmehra262001** [comment]
for any one who came up with same issue
let me tell you problem is not in the code you wrote , although I still cant point out the main issue but I'd run my own dirty tricks and can assure that its not in your written code
I exchanged my src with Patricks src and DAPP worked fine so that assures me its not my code that is creating problem
what you can try to overcome the issue is recreate react-app and move your whole code accordingly

**Comment by ikhide** [comment]
For me, I had a similar issue with Fau and Eth tokens. I realized I had a bug on the _tokenIsAllowed_ function causing a return of false for both. My function was:

`function tokenIsAllowed(address _token) public returns(bool){
        for (uint256 tokenIndex=0; tokenIndex<allowedTokens.length; tokenIndex++){
            if(allowedTokens[tokenIndex] == _token){
                return true;
            } else{
                return false;
        }
    }`

instead of:
`function tokenIsAllowed(address _token) public returns(bool){
        for (uint256 tokenIndex=0; tokenIndex<allowedTokens.length; tokenIndex++){
            if(allowedTokens[tokenIndex] == _token){
                return true;
            }
        }
        return false;
    }`
